lib/widgets/compose_box.dart

```diff
@@ -594,12 +593,9 @@ class _StreamSendButtonState extends State<_StreamSendButton> {
 
 /// The compose box for writing a stream message.
 class StreamComposeBox extends StatefulWidget {
-  const StreamComposeBox({super.key, required this.narrow, required this.streamId});
-
-  /// The narrow on view in the message list.
-  final Narrow narrow;
+  const StreamComposeBox({super.key, required this.narrow});
 
-  final int streamId;
+  final StreamNarrow narrow;
```

Comment:
In 8aec8f70b, I also considered having this widget take a `StreamNarrow` instead of a `Narrow` plus stream ID. I decided not to, because it would mean depart from the doc, which I wasn't sure we wanted to do:

```dart
/// The compose box for writing a stream message.
```

by changing the widget's use case:

- before, it was about what kind of message you could compose with it
- after, it's about how the adjacent message list is filtered (what kind of narrow is shown).

After thinking some more on this, and #149, I actually think it's a fine change to make, especially seeing your nicely simple implementation of a fixed-stream-and-topic compose box. But let's update this widget's doc to match its new purpose. We could even give the widget a more specific name, like `StreamNarrowComposeBox`, if you think that would be helpful.